### PR TITLE
Merge remote-tracking branch 'upstream/stable' into merge_stable

### PR DESCRIPTION
**Pull request by MartinNowak**

### CODE REVIEW

File: changelog/2.076.0_pre.dd [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+$(LI $(BUGZILLA 17522): win64.mak broken)
+$(LI $(BUGZILLA 17582): [REG2.059] Applying const to struct declaration should make the struct type const)
+$(LI $(BUGZILLA 17612): [REG2.063] Segmentation fault with bad object.d)
+$(LI $(BUGZILLA 17690): [REG2.066.0] scope guards leak declarations)
+$(LI $(BUGZILLA 17695): [Reg 2.076] ICE with vector negation)
+)
+$(BUGSTITLE DMD Compiler bugs,
+
+$(LI $(BUGZILLA 4014): CodeView debug type info not linked in from library)
+$(LI $(BUGZILLA 11881): -betterC switch suffers from bit rot)
```
Comment (ZombineDev) [inline changelog/2.076.0_pre.dd]: This bug is combination of many enhancement requests, only small portion of which were addressed for this release. I think we should reopen it. (The biggest unaddressed issue is that you still get link errors regarding TypeInfo, which exactly what the description of the bug talks about.)

File: changelog/2.076.0_pre.dd [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+$(LI $(BUGZILLA 17711): std.array.byPair should be usable with const AA)
+)
+$(BUGSTITLE Phobos enhancements,
+
+$(LI $(BUGZILLA 2137): Data not compressed on fly when adding to zip archive)
+$(LI $(BUGZILLA 6384): std.traits.isComparable)
+$(LI $(BUGZILLA 15759): chunks should work with only an input range)
+$(LI $(BUGZILLA 15771): FileLogger should create the output directory if it does not exist)
+$(LI $(BUGZILLA 16062): Add 'clear' method to OutBuffer $(LPAREN)std.outbuffer$(RPAREN))
+$(LI $(BUGZILLA 16744): We should have a TypeOf template so that typeof can be used with templates like staticMap)
```
Comment (ZombineDev) [inline changelog/2.076.0_pre.dd]: The addition of `TypeOf` was reverted: https://github.com/dlang/phobos/pull/5664
(Just reopened the bug.)

File: changelog/2.076.0_pre.dd [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+$(BUGSTITLE Druntime bugs,
+
+$(LI $(BUGZILLA 11594): synchronized causing segfault instead of Error.)
+$(LI $(BUGZILLA 14319): core.demangle does not support member function attributes)
+$(LI $(BUGZILLA 14563): core.demangle: Does not demangle type modifers)
+$(LI $(BUGZILLA 14576): [ddemangle] core.demangle unable to handle ambiguity in symbols)
+$(LI $(BUGZILLA 16856): D does not work on FreeBSD current $(LPAREN)what will eventually be 12$(RPAREN) due to libunwind)
+$(LI $(BUGZILLA 17609): core.demangle demangles delegate variables as functions)
+$(LI $(BUGZILLA 17610): core.demangle shows return type of template alias parameter)
+$(LI $(BUGZILLA 17611): core.demangle cannot demangle delegates with function attributes)
```
Comment (ZombineDev) [inline changelog/2.076.0_pre.dd]: Do these mangling related bug fixes depend on Rainer's unmerged dmd PR, or the merged druntime PR is sufficient?

**Comment by MartinNowak** [comment]
Let's just reopen the marked bugs, we'll later regenerate the changelog anyhow.

**Pull request merged**